date.go

```diff
@@ -83,6 +83,25 @@ func (d Date) AddDate(years int, months int, days int) Date {
 	return DateFromStdTime(d.t.AddDate(years, months, days))
 }
 
+// AddMonthsNoOverflow adds a month to the current time, not overflowing in case the
+// destination month has less days than the current one.
+// Positive value travels forward while negative value travels into the past.
+func (d Date) AddMonthsNoOverflow(m int) Date {
+	addedDate := d.AddDate(0, m, 0)
+	if d.Day() != addedDate.Day() {
+		d = addedDate
+		return d.PreviousMonthLastDay()
```

Comment:
odd bit of code. why not `return addedDate.PreviousMonthLastDay()`?

Reply:
Changed